unlock-app/src/components/creator/FatalError.tsx

```diff
@@ -120,39 +130,52 @@ export const MissingAccount = () => (
 )
 
 export const ContractNotDeployed = () => (
-  <DefaultError title="Unlock not deployed">
+  <DefaultError
+    title="Unlock not deployed"
+    illustration="/static/images/illustrations/wallet.svg"
+    critical
+  >
     <p>The Unlock contract has not been deployed at the configured address.</p>
   </DefaultError>
 )
 
 export const NotEnabledInProvider = () => (
-  <DefaultError title="Not enabled in provider">
+  <DefaultError
+    title="Not enabled in provider"
+    illustration="/static/images/illustrations/wallet.svg"
+    critical
+  >
     <p>You did not approve Unlock in your web3 wallet.</p>
   </DefaultError>
 )
 
-export const mapping = {
+type Template = () => JSX.Element
+
+interface Mapping {
+  [key: string]: Template
+}
+export const mapping: Mapping = {
   FATAL_MISSING_PROVIDER: MissingProvider,
   FATAL_NO_USER_ACCOUNT: MissingAccount,
-  FATAL_WRONG_NETWORK: WrongNetwork,
   FATAL_NON_DEPLOYED_CONTRACT: ContractNotDeployed,
   FATAL_NOT_ENABLED_IN_PROVIDER: NotEnabledInProvider,
-  '*': DefaultError,
+  '*': FallbackError,
 }
 
 export function mapErrorToComponent(
-  error,
-  errorMetadata,
-  overrideMapping = {}
+  error: FatalError,
+  overrideMapping: Mapping = {}
 ) {
-  // if the error condition exists, set it to the mapped fatal error component
-  // or to the fallback
-  // if no error exists, set it to false
-  const Error = error
-    ? overrideMapping[error] || mapping[error] || mapping['*']
-    : false
-
-  return Error && <Error {...errorMetadata} />
+  const { message } = error
+  if (message === 'FATAL_WRONG_NETWORK') {
```

Comment:
This is a special case -- it's the only error that carries the separate data payload for the time being. I'd like a cleaner solution to this, but I think it may be out of scope for this PR and this is good enough to hold us for a while.